### ISSUE TITLE
Exiting when on a secondary branch

### ISSUE THREAD

**Issue by llakala** [quatquatt/RebuildButLessDumb#3]
I thought I'd fixed this. However, I'd actually only made a wrapping error, that's easier to deal with.

See, there are four cases. 
- No uncommitted changes, on main/master
- Uncommitted changes, on main/master
- No uncommitted changes, on a secondary branch
- Uncommitted changes, on a secondary branch

Currently, both the 3rd and 4th cases, where we're on a secondary branch, mean that `unify` literally just doesn't work. I get this output: 
![image](https://github.com/user-attachments/assets/a5bbf9ac-f870-4d53-a144-7d51cc8ca3e5)
You see that? No? It's because it's outputting nothing.

This was happening before, and I thought I'd fixed it. But I'd actually just *reversed* the error. Before, this would happen whenever I was on a primary branch. But I swapped the return values within `on_secondary_branch`. Now it returns the opposite of what it should. Which is *bad*,  but it means that `unify` works perfectly on main, it just fails on other branches. 

TLDR: when we're on another branch, we seem to be exiting early at some point. We need to add lots of logging to find out exactly where this is.

**Comment by llakala** [comment]
Fixed with 8d234b9b3fec8ef91a5c9212b8960f678280def7. Before, we were iterating over lists like this: 
```sh
PRIMARY_BRANCHES=("main" "master")
for primary_branch in "''${PRIMARY_BRANCHES[@]}"; do
```
But this was apparently causing some weird behavior, leading to something like this being done (logs from `set -x`): 
```sh
+ for primary_branch in "${PRIMARY_BRANCHES[@]}"
+ [[ main == \m\a\i\n ]]`
```
Yeah. Weird. Now we just have a string like this: 
"main master" 
And we're iterating over it. Works perfectly.
